packages/components/src/color-picker/alpha.js

```diff
@@ -34,34 +34,62 @@ import { noop } from 'lodash';
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { Component, createRef } from '@wordpress/element';
+import { useState, useEffect, useRef } from '@wordpress/element';
 import { TAB } from '@wordpress/keycodes';
-import { pure } from '@wordpress/compose';
 
 /**
  * Internal dependencies
  */
 import { calculateAlphaChange } from './utils';
 import KeyboardShortcuts from '../keyboard-shortcuts';
 
-export class Alpha extends Component {
-	constructor() {
-		super( ...arguments );
+export default function Alpha( { rgb, hsl, onChange = noop } ) {
+	const [ mouseDown, setMouseDown ] = useState( false );
+	const container = useRef();
 
-		this.container = createRef();
-		this.increase = this.increase.bind( this );
-		this.decrease = this.decrease.bind( this );
-		this.handleChange = this.handleChange.bind( this );
-		this.handleMouseDown = this.handleMouseDown.bind( this );
-		this.handleMouseUp = this.handleMouseUp.bind( this );
+	const rgbString = `${ rgb.r },${ rgb.g },${ rgb.b }`;
+	const gradient = {
+		background: `linear-gradient(to right, rgba(${ rgbString }, 0) 0%, rgba(${ rgbString }, 1) 100%)`,
+	};
+	const pointerLocation = { left: `${ rgb.a * 100 }%` };
+
+	function handleChange( e ) {
+		const change = calculateAlphaChange( e, { hsl }, container.current );
+		if ( change ) {
+			onChange( change, e );
+		}
+	}
+
+	function handleMouseUp() {
+		setMouseDown( false );
 	}
 
-	componentWillUnmount() {
-		this.unbindEventListeners();
+	function preventKeyEvents( event ) {
+		if ( event.keyCode === TAB ) {
+			return;
+		}
+		event.preventDefault();
```

Comment:
Any idea why these events must be default prevented? Since you're rewriting the code, any insights you might have about unusual undocumented code might be valuable to document or comment on.

Reply:
I'm not sure why this code was used before. it's just preventing keydown. I've to test the component without this listener.